packages/vue/src/components/atoms/SfInput/SfInput.vue

```diff
@@ -155,6 +199,50 @@ export default {
     isPassword() {
       return this.type === "password" && this.hasShowPassword;
     },
+    computedMessageSlotName() {
```

Comment:
There are the same logic if-else statements. I think there are 2 solutions you can consider.
1. Create one computet property that returns an object like `if (this.errorMessage && !this.valid) {
        return {slotName: "show-error-message", message: this.errorMessage, messageClass: "sf-input__message--error"}}`
2. Create a function to which you pass parameters `superFunction(error, success, hint) {
 if (this.errorMessage && !this.valid) {
        return error;
      } else if (this.successMessage && this.valid) {
        return success;
      } else if (this.hintMessage) {
        return this.required ? hint : "";
      } else {
        return "";
      }
}`
And then you can just `
computedMessageSlotName() {
return superFunction("show-error-message","show-success-message", "show-hint-message")
},
    computedMessage() {
return superFunction(this.errorMessage,this.successMessage, this.hintMessage)
    },
    computedMessageClass() {
return superFunction("sf-input__message--error","sf-input__message--success", "sf-input__message--success")
    },`

Wdyt?

Reply:
of course you are right, maybe I'll try second solution.